csharp/extractor/Semmle.Extraction.CSharp/SymbolExtensions.cs

```diff
@@ -434,6 +524,13 @@ public static bool IsSourceDeclaration(this IParameterSymbol parameter)
             return true;
         }
 
+        /// <summary>
+        /// Gets the base type of `symbol`. Unlike `symbol.BaseType`, this excludes effective base
+        /// types of type parameters as well as `object` base types.
+        /// </summary>
+        public static INamedTypeSymbol GetNonObjectBaseType(this ITypeSymbol symbol, Context cx) =>
```

Comment:
Are we excluding `object` because we can add it back in QL? And we want to reduce the size of the trap file/DB?

Reply:
Actually, this is strictly speaking not needed, as we are now also adding base classes to TRAP IDs for disambiguation.

Reply:
But yes, since we are now also adding base classes to the IDs, only doing it for non-`object` base classes reduces TRAP size.